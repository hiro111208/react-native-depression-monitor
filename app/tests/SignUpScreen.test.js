import React from "react";
import renderer from "react-test-renderer";
import SignUpScreen from "../screens/SignUpScreen";

import * as firebase from "../../node_modules/@firebase/testing";

describe("Testing SignUpScreen.js", () => {

    beforeAll(() => {
        jest.setTimeout(100); //find appropriate timeout after testing
    });

    it("SignUpScreen renders correctly", () => {
        const tree = renderer.create(<SignUpScreen />).toJSON();
        expect(tree).toMatchSnapshot();
    });

    test("Sign Up should work with correct credentials and verification", async () => {

    });

    test("Sign Up should not work with correct credentials but no verification", async () => {

    });

    test("Sign Up should not work with incorrect email", async () => {

    });

    test("Sign Up should not work with incorrect email", async () => {

    });

});

// test("signup should work with correct credentials and verification", () => 
// {
//     const user = createUserWithEmailAndPassword("success@signup.com", "password");
//     //verify the user
//     expect(user.user).toBeTruthy();
//     expect(isVerified()).toBe(true);
//     //test for email and password value?
// });

// test("signup should not work with correct credentials but no verification", () => 
// {
//     let error = "";
//     try {
//         createUserWithEmailAndPassword("fail@signup.com", "password");
//     } catch (err) {
//         error = err.toString();
//     }
//     expect(error).toEqual(`Please verify your email through the link we"ve sent to: fail@signup.com`);
// });

// test("signup should work not work with invalid email", () => 
// {
//     let error = "";
//     try {
//          createUserWithEmailAndPassword("fail@signup", "password");
//     } catch (err) {
//         error = err.toString();
//     }
//     expect(error).toEqual(`failed to send email verification`);
// });

//test with email that has already been signed up should fail
//The email address is already in use by another account.

// test("signup should not work with password shorter than 6 characters", async () => 
// {
//     let error = "";
//     try {
//         await createUserWithEmailAndPassword("tooshort@login.com", "short");
//     } catch (err) {
//         error = err.toString();
//     }

//     expect(error).toEqual("");//error.message?
// });

// test("signup should not work with password longer than 15 characters", async () => 
// {
//     let error = "";
//     try {
//         await createUserWithEmailAndPassword("toolong@login.com", "thispasswordistoolong");
//     } catch (err) {
//         error = err.toString();
//     }

//     expect(error).toEqual("");//error.message?
// });