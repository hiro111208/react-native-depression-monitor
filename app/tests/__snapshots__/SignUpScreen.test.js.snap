// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing SignUpScreen.js SignupScreen renders correctly 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#fff",
      "display": "flex",
      "flex": 1,
      "flexDirection": "column",
      "justifyContent": "center",
      "padding": 35,
    }
  }
>
  <TextInput
    allowFontScaling={true}
    onChangeText={[Function]}
    placeholder="First Name"
    rejectResponderTermination={true}
    style={
      Object {
        "alignSelf": "center",
        "borderBottomWidth": 1.5,
        "borderColor": "#ccc",
        "marginBottom": 15,
        "marginTop": 15,
        "paddingBottom": 20,
        "width": "100%",
      }
    }
    underlineColorAndroid="transparent"
    value=""
  />
  <TextInput
    allowFontScaling={true}
    keyboardType="email-address"
    onChangeText={[Function]}
    placeholder="Email"
    rejectResponderTermination={true}
    style={
      Object {
        "alignSelf": "center",
        "borderBottomWidth": 1.5,
        "borderColor": "#ccc",
        "marginBottom": 15,
        "marginTop": 15,
        "paddingBottom": 20,
        "width": "100%",
      }
    }
    testID="TEST_ID_EMAIL_INPUT"
    underlineColorAndroid="transparent"
    value=""
  />
  <View
    style={
      Array [
        Object {
          "alignSelf": "center",
          "borderBottomWidth": 1.5,
          "borderColor": "#ccc",
          "marginBottom": 15,
          "marginTop": 15,
          "paddingBottom": 20,
          "width": "100%",
        },
        Object {
          "alignItems": "center",
          "flexDirection": "row",
          "justifyContent": "space-between",
        },
      ]
    }
  >
    <TextInput
      allowFontScaling={true}
      autoCorrect={false}
      maxLength={15}
      onChangeText={[Function]}
      placeholder="Password"
      rejectResponderTermination={true}
      secureTextEntry={true}
      testID="TEST_ID_PASSWORD_INPUT"
      underlineColorAndroid="transparent"
      value=""
    />
    <View
      collapsable={false}
    >
      <View
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
          }
        }
      >
        <View
          style={
            Object {
              "flexDirection": "row",
              "justifyContent": "flex-end",
              "textAlign": "right",
            }
          }
        >
          <Image
            resizeMode="contain"
            source={
              Object {
                "testUri": "../../../app/assets/question_mark.png",
              }
            }
            style={
              Object {
                "height": 22,
              }
            }
          />
        </View>
      </View>
      <Modal
        animationType="fade"
        hardwareAccelerated={false}
        onShow={[Function]}
        transparent={true}
        visible={false}
      >
        <View
          accessible={true}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "backgroundColor": "transparent",
              "flex": 1,
              "opacity": 1,
            }
          }
        >
          <View>
            <View
              style={
                Object {
                  "backgroundColor": "transparent",
                  "height": 0,
                  "left": 0,
                  "overflow": "visible",
                  "position": "absolute",
                  "top": 0,
                  "width": 0,
                }
              }
            >
              <View
                style={
                  Object {
                    "flexDirection": "row",
                    "justifyContent": "flex-end",
                    "textAlign": "right",
                  }
                }
              >
                <Image
                  resizeMode="contain"
                  source={
                    Object {
                      "testUri": "../../../app/assets/question_mark.png",
                    }
                  }
                  style={
                    Object {
                      "height": 22,
                    }
                  }
                />
              </View>
            </View>
            <View
              style={
                Object {
                  "left": -7.5,
                  "position": "absolute",
                  "top": -2,
                }
              }
            >
              <View
                style={
                  Object {
                    "backgroundColor": "transparent",
                    "borderBottomColor": "#ffa351ff",
                    "borderBottomWidth": 15,
                    "borderLeftColor": "transparent",
                    "borderLeftWidth": 8,
                    "borderRightColor": "transparent",
                    "borderRightWidth": 8,
                    "borderStyle": "solid",
                    "height": 0,
                    "width": 0,
                  }
                }
              />
            </View>
            <View
              style={
                Object {
                  "alignItems": "center",
                  "backgroundColor": "#ffa351ff",
                  "borderRadius": 10,
                  "display": "flex",
                  "flex": 1,
                  "height": 80,
                  "justifyContent": "center",
                  "left": -17.999,
                  "padding": 10,
                  "position": "absolute",
                  "top": 10,
                  "width": 300,
                }
              }
              testID="tooltipPopoverContainer"
            >
              <Text
                style={
                  Object {
                    "color": "white",
                  }
                }
              >
                Password must be 6 to 15 characters long. Can contain any alphaneumeric or special character.
              </Text>
            </View>
          </View>
        </View>
      </Modal>
    </View>
  </View>
  <Text
    style={
      Object {
        "color": "red",
      }
    }
  >
    
  </Text>
  <Text
    style={
      Object {
        "color": "red",
      }
    }
  >
    
  </Text>
  <View
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignSelf": "center",
        "backgroundColor": "#ffa351ff",
        "borderRadius": 50,
        "marginTop": 10,
        "opacity": 1,
        "paddingBottom": 15,
        "paddingTop": 15,
        "width": 300,
      }
    }
    testID="TEST_ID_SIGNUP_BUTTON"
  >
    <Text
      style={
        Object {
          "color": "white",
          "fontSize": 15,
          "textAlign": "center",
        }
      }
    >
      SIGNUP
    </Text>
  </View>
  <Text
    onPress={[Function]}
    style={
      Object {
        "color": "#ffa351ff",
        "marginTop": 25,
        "textAlign": "center",
      }
    }
  >
    Already Registered? Click here to login
  </Text>
</View>
`;
